app/Actions/Photo/Pipes/Init/FindLivePartner.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Actions\Photo\Pipes\Init;
+
+use App\Contracts\PhotoCreate\InitPipe;
+use App\DTO\PhotoCreate\InitDTO;
+use App\Exceptions\Internal\IllegalOrderOfOperationException;
+use App\Exceptions\Internal\LycheeAssertionError;
+use App\Image\Files\BaseMediaFile;
+use App\Models\Photo;
+
+/**
+ * Assert wether we support said file.
+ */
+class FindLivePartner implements InitPipe
+{
+	/**
+	 * {@inheritDoc}
+	 */
+	public function handle(InitDTO $state, \Closure $next): InitDTO
+	{
+		try {
+			// find a potential partner which has the same content id
+			if ($state->exifInfo->livePhotoContentID !== null) {
+				$state->livePartner = Photo::query()
+					->where('live_photo_content_id', '=', $state->exifInfo->livePhotoContentID)
+					->where('album_id', '=', $state->album?->id)
+					->whereNull('live_photo_short_path')->first();
+			}
+
+			// if a potential partner has been found, ensure that it is of a
+			// different kind then the uploaded media.
+			if (
+				$state->livePartner !== null && !(
+					BaseMediaFile::isSupportedImageMimeType($state->exifInfo->type) && $state->livePartner->isVideo() ||
+					BaseMediaFile::isSupportedVideoMimeType($state->exifInfo->type) && $state->livePartner->isPhoto()
+				)
+			) {
+				$state->livePartner = null;
+			}
```

Comment:
Isn't this saying "If it's different, set to null"?

Reply:
Indeed, `isVideo()` in the end is doing `BaseMediaFile::isSupportedVideoMimeType($state->livePartner->type)` but with an additional null check.
And similarly `isPhoto()` is doing `BaseMediaFile::isSupportedPhotoMimeType($state->livePartner->type)` but with an additional null check.

But this makes the checks more stricts as in it enforces Photo-Video pairing.

Reply:
But isn't the goal for them to be different? i.e. the conditions here are the "good" conditions?

Reply:
I think you are missing the sneaky `!(` at line 34. No?

Reply:
Yes. Yes I was! :laughing: